modules/percolator/src/main/java/org/elasticsearch/percolator/QueryAnalyzer.java

```diff
@@ -341,9 +353,54 @@ public static Result analyze(Query query) {
         };
     }
 
+    static Function<Query, Result> pointRangeQuery() {
+        return query -> {
+            PointRangeQuery pointRangeQuery = (PointRangeQuery) query;
+            // We need to check the number type before indexing to ensure we support it.
+            // hacky, but otherwise we need to use the MapperService to determine what kind of number field this is
+            Class<?> enclosingClass = pointRangeQuery.getClass().getEnclosingClass();
+            final byte[] lowerPoint = pointRangeQuery.getLowerPoint();
+            final byte[] upperPoint = pointRangeQuery.getUpperPoint();
+            final RangeType rangeType;
+            final Number width;
+            if (LongPoint.class.isAssignableFrom(enclosingClass)) {
```

Comment:
I like this approach. However I tried to remove this, but I ran in an issue. Lets say there are two ranges queries, int interval between 10 and 15 and long interval between 10 and 20. Comparing the `byte[]` from `NumericUtils#subtract(...)` results in the long range being picked instead of the int range. This is unexpected since the int range is the smallest interval.  You know a way around this problem, while still being able to use the `NumericUtils#subtract(...)` approach? I did make the code cleaner.